airflow/contrib/auth/backends/google_auth.py

```diff
@@ -109,6 +109,7 @@ def login(self, request):
         return self.google_oauth.authorize(callback=url_for(
             'google_oauth_callback',
             _external=True,
+            _scheme=get_config_param('scheme'),
```

Comment:
Shouldn't redirect URL always be on https? I would prefer hard code _schem="https".

Reply:
Not necessarily. It's valid to pass an http URL. Perhaps not advisable, but valid behaviour.

Reply:
Agree, but it's probably a bad practice from a security perspective and may end up hurting everyone involved.

Reply:
Well, it's not my decision to make. I would be comfortable making it https-only, but I tried to be less opinionated in this patch to make it easier to swallow. If it can land as https-only I'm happy to change it.

Reply:
Since this is google auth, would prefer https. 
@criccomini for a second opinion.

Reply:
I checked, and there's no default for google.scheme in the config file, so the operator will be forced to decide. If they don't make a choice, they'll see this error:

```
[2018-01-03 10:14:10,292] {configuration.py:206} WARNING - section/key [google/scheme] not found in config
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "airflow/configuration.py", line 408, in get
    return conf.get(section, key, **kwargs)
  File "airflow/configuration.py", line 211, in get
    "in config".format(**locals()))
airflow.exceptions.AirflowConfigException: section/key [google/scheme] not found in config
```

I can see arguments for both ways. I think I'm partial to forcing HTTPS, though, since this is security-related. @intellectronica, can you update the PR?

Reply:
@criccomini @fenglu-g updated to always use http.